src/autogluon/cloud/predictor/cloud_predictor.py

```diff
@@ -792,17 +812,43 @@ def predict_real_time(self, test_data, accept="application/x-parquet"):
 
         Returns
         -------
-        Pandas.DataFrame
-        Predict results in DataFrame
+        Pandas.Series
+        Predict results in Series
         """
-        assert self.endpoint, "Please call `deploy()` to deploy an endpoint first."
-        assert accept in VALID_ACCEPT, f"Invalid accept type. Options are {VALID_ACCEPT}."
-        if type(test_data) == str:
-            test_data = load_pd.load(test_data)
-        if not isinstance(test_data, pd.DataFrame):
-            raise ValueError("test_data must be either a pandas.DataFrame, a local path or a s3 path")
+        self._validate_predict_real_time_args(accept)
+        test_data = self._load_predict_real_time_test_data(test_data)
+        pred, _ = self._predict_real_time(test_data=test_data, accept=accept)
+
+        return pred
+
+    def predict_proba_real_time(self, test_data, accept="application/x-parquet"):
+        """
+        Predict probability with the deployed SageMaker endpoint. A deployed SageMaker endpoint is required.
+        This is intended to provide a low latency inference.
+        If you want to inference on a large dataset, use `predict_proba()` instead.
+        If your problem_type is regression, this functions identically to `predict_real_time`, returning the same output.
+
+        Parameters
+        ----------
+        test_data: Union(str, pandas.DataFrame)
+            The test data to be inferenced. Can be a pandas.DataFrame, or a local path to csv file.
+        accept: str, default = application/x-parquet
+            Type of accept output content.
+            Valid options are application/x-parquet, text/csv, application/json
+
+        Returns
+        -------
+        Pandas.DataFrame or Pandas.Series
+            Will return a Pandas.Series when it's a regression problem. Will return a Pandas.DataFrame otherwise
```

Comment:
Why not instead crash if regression?

Reply:
user should call predict for regression, not predict_proba

Reply:
Note: Currently Tabular will return predict output for regression when called via predict_proba, so maybe we should do the same for now in Cloud, however probably we want to error starting in AutoGluon v1.0.

Reply:
Yea. I checked the behavior of TabularPredictor and decided to follow. We can change the behavior to be in sync with tabular when it's changed in the future

Reply:
Created a GitHub issue to start erroring in v0.8 release: https://github.com/autogluon/autogluon/pull/2684